backend/app/adapter/db/url.go

```diff
@@ -78,6 +79,47 @@ WHERE "%s"=$1;`,
 	return url, nil
 }
 
+// GetURLByVisibility implements retrieval of URL associated with userEmail based on type specified
+// TODO: Add urlType based querying after column addition
+func (u URLSql) GetURLByVisibility(userEmail, urlType string) ([]search.URLSearchResponse, error) {
```

Comment:
Let's return a slice of urls. URL repository should have knowledge of `search` related knowledge.

Reply:
Please replace `userEmail` with `user entity.User`

Reply:
This function is about querying the relationship between the user and the urls. It should be renamed to `GetAliasByUser(user entity.User) ([]string, error)` and be moved into https://github.com/byliuyang/short/blob/master/backend/app/adapter/db/userurl.go

Reply:
If we just return the list of URLs, how will we differentiate URLTypes if we query just for user-related URLs? i.e when API called without any queryType: public or private.

Reply:
@lucifercr07 Let's add `GetURLs()` method for `UserURLRelationSQL` struct instead.